doc/source/train/user-guides/results.rst

```diff
@@ -0,0 +1,91 @@
+Inspecting Training Results
+===========================
+
+The return value of your :meth:`Trainer.fit() <ray.train.trainer.BaseTrainer.fit>`
+call is a :class:`~ray.air.result.Result` object.
+
+The :class:`~ray.air.result.Result` object contains, among others:
+
+- The last reported metrics (e.g. the loss)
+- The last reported checkpoint (to load the model)
+- Error messages, if any errors occurred
+
+Last reported metrics
+---------------------
+
+Use :py:attr:`Result.metrics <ray.air.result.Result.metrics>` to retrieve the
+latest reported metrics.
+
+This corresponds to the metrics you passed to :func:`train.report <ray.train.report>`
+as an argument :ref:`in your training function <train-monitoring-and-logging>`.
+
+.. literalinclude:: ../doc_code/key_concepts.py
+    :language: python
+    :start-after: __result_metrics_start__
+    :end-before: __result_metrics_end__
+
+Dataframe of all reported metrics
+---------------------------------
+Use :py:attr:`Result.metrics_dataframe <ray.air.result.Result.metrics_dataframe>` to retrieve
+a pandas DataFrame of all reported metrics.
+
+.. literalinclude:: ../doc_code/key_concepts.py
+    :language: python
+    :start-after: __result_dataframe_start__
+    :end-before: __result_dataframe_end__
+
+
+Last saved checkpoint
```

Comment:
Same as above but for checkpoints, there is a difference in ways to retrieve checkpoints, but should have a common section for what the user might do with the checkpoints.